src/mmw/apps/geoprocessing_api/views.py

```diff
@@ -224,11 +227,41 @@ def start_rwd(request, format=None):
     """
     user = request.user if request.user.is_authenticated() else None
     created = now()
-    location = request.data['location']
+    location = request.data.get('location')
+
+    def validate_location(loc):
+        if loc is None or type(loc) is not list or len(loc) is not 2:
+            return False
+        else:
+            [lat, lng] = loc
+            return type(lat) is float and type(lng) is float
+
+    if not validate_location(location):
+        error = ('Invalid required `location` parameter value '
+                 '`{}`. Must be a `[lat, lng]`.'.format(location))
+        raise ValidationError(error)
+
     data_source = request.data.get('dataSource', 'drb')
+
+    if data_source not in ['drb', 'nhd']:
+        error = ('Invalid optional `dataSource` parameter value '
+                 '`{}`. Must be `drb` or `nhd`.'.format(data_source))
+        raise ValidationError(error)
+
     snapping = request.data.get('snappingOn', False)
+
+    if type(snapping) is not bool:
+        error = ('Invalid optional `snappingOn` parameter value '
+                 '`{}`. Must be `true` or `false`.'.format(snapping))
+        raise ValidationError(error)
+
     simplify = request.data.get('simplify', False)
 
+    if type(simplify) is not bool and not isinstance(simplify, Number):
```

Comment:
Should this be
```py
if simplify is not None and not isinstance(simplify, Number):
```

Reply:
The get-or-else line above assigns it to `False` currently if there's no supplied value, so it should always be either `False` or a number. This also means clients could potentially pass `false` or `true` as the value and have it succeed -- not sure whether this would have adverse effects?

Reply:
Ah, okay. Maybe it should check if `simplify is False` then?

Reply:
TIL:

```py
>>> isinstance(True, Number)
True
```

Reply:
Pushed some changes to address this. I rebased on `develop`, so went ahead and squashed the fixup too.